shipyard_blueprints/sqlserver/shipyard_sqlserver/cli/upload_file.py

```diff
@@ -63,122 +66,64 @@ def get_args():
+            folder_name=dir_name, file_name=file_name
+        )
+        table_name = args.table_name
+        insert_method = args.insert_method
+
+        client = SqlServerClient(
+            user=args.username,
+            pwd=args.password,
+            host=args.host,
+            database=args.database,
+            port=args.port,
+            url_params=args.url_parameters,
         )
-    print(f"{source_full_path} successfully uploaded to {table_name}.")
+        client.connect()
+        logger.info("Successfully connected to SQL Server")
+
+        if match_type == "regex_match":
+            file_names = shipyard.files.find_all_local_file_names(dir_name)
+            matching_file_names = shipyard.files.find_all_file_matches(
+                file_names, re.compile(file_name)
+            )
+            logger.info(
+                f"{len(matching_file_names)} files found. Preparing to upload..."
+            )
 
+            for index, file in enumerate(matching_file_names):
+                df = pd.read_csv(file)
+                if index > 0:
+                    insert_method = "append"
+                client.upload(df=df, table_name=table_name, insert_method=insert_method)
+                logger.info(f"Upload of {file} complete")
 
-def main():
-    args = get_args()
-    source_file_name_match_type = args.source_file_name_match_type
-    source_file_name = args.source_file_name
-    source_folder_name = args.source_folder_name
-    source_full_path = combine_folder_and_file_name(
-        folder_name=source_folder_name, file_name=source_file_name
-    )
-    table_name = args.table_name
-    insert_method = args.insert_method
+            logger.info(f"Successfully loaded all files to {table_name}")
 
-    db_string = create_connection_string(args)
-    try:
-        db_connection = create_engine(db_string, fast_executemany=True)
-    except Exception as e:
-        print(f"Failed to connect to database {args.database}")
-        raise (e)
+        else:
+            df = pd.read_csv(file_path)
+            client.upload(df=df, table_name=table_name, insert_method=insert_method)
+            logger.info(f"Successfully loaded {file_path} to {table_name}")
 
-    if source_file_name_match_type == "regex_match":
-        file_names = find_all_local_file_names(source_folder_name)
-        matching_file_names = find_all_file_matches(
-            file_names, re.compile(source_file_name)
-        )
-        print(f"{len(matching_file_names)} files found. Preparing to upload...")
-
-        for index, key_name in enumerate(matching_file_names):
-            upload_data(
-                source_full_path=key_name,
-                table_name=table_name,
-                insert_method=insert_method,
-                db_connection=db_connection,
-            )
+    except ExitCodeException as ec:
+        logger.error(ec.message)
+        sys.exit(ec.exit_code)
 
-    else:
```

Comment:
Should also handle file not found error
`EXIT_CODE_FILE_NOT_FOUND` for better reporting